packages/components/src/public/QueryModel/GridPanel.tsx

```diff
@@ -813,9 +813,26 @@ export class GridPanel<T = {}> extends PureComponent<Props<T>, State> {
 
         return new Promise((resolve, reject) => {
             const view = queryInfo?.getView(viewName, true);
-
-            const updatedViewInfo = view.mutate({
-                // update/set sorts and filters to combine view and user defined items
+            let updatedViewInfo = view;
+            if (view.isDefault && !view.session) {
+                let columns = view.columns;
+                const columnFieldKeys = view.columns.map(col => {
+                    return col.fieldKey.toLowerCase()
+                }).toArray();
+                queryInfo.columns.forEach(queryCol => {
+                    if (queryCol.fieldKey && queryCol.addToSystemView && columnFieldKeys.indexOf(queryCol.fieldKey.toLowerCase()) === -1) {
+                        columns = columns.push({
+                            fieldKey: queryCol.fieldKey,
+                            key: queryCol.fieldKey,
+                            name: queryCol.name,
+                            title: queryCol.caption,
+                        });
+                    }
+                });
```

Comment:
could this be moved to be a method off of the model? something like `model.getAddToSystemViewColumns()`. That would likely make this easier to jest test as well.

Reply:
Yeah, I was struggling a bit with where to put this. I've added a method to `ViewInfo` as that seems most appropriate. Let me know if that doesn't seem right to you.

Reply:
Looks good